e2e/cypress/integration/messaging/permalink_message_edit_spec.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// [number] indicates a test step (e.g. # Go to a page)
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Permalink message edit', () => {
+    it('M18717 - Edit a message in permalink view', () => {
+        // # Login as "user-1" and go to /
+        cy.apiLogin('user-1');
+        cy.visit('/');
+
+        const searchWord = 'searchtest';
+
+        // # Post message
+        cy.postMessage(searchWord);
+
+        // # Search for searchWord
+        cy.get('#searchBox').type(searchWord).type('{enter}').then(() => {
+            // # Jump to permalink view
+            cy.get('.search-item__jump').first().click();
+
+            cy.getLastPostId().then((postId) => {
+                // # Click on ... button of last post matching the searchWord
+                cy.clickPostDotMenu(postId);
+
+                // # Click on edit post
+                cy.get(`#edit_post_${postId}`).click();
+
+                // # Add new text in edit box
+                cy.get('#edit_textbox').clear().type('edited searchtest');
+
+                // # Click edit button
+                cy.get('#editButton').click();
+
+                // # Check edited post
+                cy.get(`#postMessageText_${postId}`).should('contain', 'edited searchtest');
+                cy.get(`#postEdited_${postId}`).should('contain', '(edited)');
```

Comment:
Please extract this into a function and use it as well to verify for user 2.
```suggestion
                verifyEditedPermalink(postId, editedText);
```

Such can be
```javascript
function verifyEditedPermalink(postId, text) {
    // * Check that it redirected to permalink URL
    cy.url().should('include', `/ad-1/pl/${postId}`);

    // * Verify edited post
    cy.get(`#postMessageText_${postId}`).should('have.text', text);
    cy.get(`#postEdited_${postId}`).should('have.text', '(edited)');
}
```

Note that I added quick check to URL just to ensure it's really on permalink view.

Reply:
I like this, it looks cleaner and ensures the user is still in permalink view.  Thank you for your feedback